source/fundamentals/data-formats/pojo-customization.txt

```diff
@@ -10,78 +10,653 @@ POJO Customization
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Field Name
+     - Description
+
+   * - Name
+     - The POJO class name to associate with the ``ClassModel``.
 
-- Description
-- Table of properties
-- Link to Conventions section for example of ClassModelBuilder
+   * - InstanceCreatorFactory
+     - Contains a new instance factory that creates new instances of the POJO.
+       By default, it requires the POJO to have an empty constructor.
+
+   * - PropertyModels
+     - Contains a list of ``PropertyModel`` instances that specify how to
+       convert data to and from BSON for a field in the POJO.
+
+   * - IdPropertyModelHolder
+     - Specifies the POJO field that corresponds to the document id field. Optional.
+
+   * - Discriminator Key
+     - Specifies the name of the discriminator field. Optional.
+
+   * - Discriminator Value
+     - A lookup value that represents the POJO class. Optional.
+
+   * - Discriminator Flag
+     - Specifies whether to serialize the discriminator, off by default. Optional.
+
+For more information on this class, see the :java-docs:`ClassModel </apidocs/bson/org/bson/codecs/pojo/ClassModel.html>`
+API documentation.
+
+To instantiate a ``ClassModel`` use the ``ClassModel.builder()`` method and
+specify your POJO class. The builder uses reflection to create the required
+metadata.
+
+.. code-block:: java
+   :copyable: true
+
+   ClassModel<Flower> classModel = ClassModel.builder(Flower.class).build();
+
+.. _property-model:
 
 PropertyModel
--------------
+~~~~~~~~~~~~~
+
+A ``PropertyModel`` stores information on how to serialize/deserialize a
+specific field in a document.
+
+The ``PropertyModel`` contains the following information:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Field Name
+     - Description
+
+   * - Name
+     - Specifies the property name in the model.
+
+   * - Read Name
+     - Name of the property to use as the key when serializing to BSON.
+
+   * - Write Name
+     - Name of the property to use as the key when deserializing from BSON.
+
+   * - Type data
+     - Contains an instance of ``org.bson.codecs.pojo.TypeData`` that
+       describes the data type for the field.
+
+   * - Codec
+     - Specifies a codec to use to encode or decode the field. Optional.
+
+   * - Serialization checker
+     - Determines whether to serialize a value by using the criteria
+       specified in the checker.
 
-- Description
-- Table of properties
-- Link to Conventions section for example of PropertyModelBuilder
+   * - Property accessor
+     - Method used to access the value of the property from the POJO
+
+   * - useDiscriminator
+     - Specifies whether to use the discriminator.
+
+To create a ``PropertyModel`` use a ``PropertyModelBuilder`` which you
+can instantiate by calling the ``PropertyModel.builder()`` method.
+
+For more information, see the
+:java-docs:`PropertyModel.Builder API documentation <apidocs/bson/org/bson/codecs/pojo/PropertyModelBuilder.html>`.
```

Comment:
```suggestion
:java-docs:`PropertyModel.Builder <apidocs/bson/org/bson/codecs/pojo/PropertyModelBuilder.html>` API documentation.
```
The other links to API docs don't include "API documentation" in the link itself.